sdk/storage/azure-storage-file-datalake/src/samples/java/com/azure/storage/file/datalake/BasicExample.java

```diff
@@ -0,0 +1,132 @@
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.Locale;
+
+/**
+ * This example shows how to start using the Azure Storage Data Lake SDK for Java.
+ */
+public class BasicExample {
+
+    /**
+     * Entry point into the basic examples for Storage datalake.
+     *
+     * @param args Unused. Arguments to the program.
+     * @throws IOException If an I/O error occurs
+     * @throws RuntimeException If the downloaded data doesn't match the uploaded data
+     */
+    public static void main(String[] args) throws IOException {
+
+        /*
+         * From the Azure portal, get your Storage account's name and account key.
+         */
+        String accountName = SampleHelper.getAccountName();
+        String accountKey = SampleHelper.getAccountKey();
+
+        /*
+         * Use your Storage account's name and key to create a credential object; this is used to access your account.
+         */
+        StorageSharedKeyCredential credential = new StorageSharedKeyCredential(accountName, accountKey);
+
+        /*
+         * From the Azure portal, get your Storage account dfs service URL endpoint.
+         * The URL typically looks like this:
+         */
+        String endpoint = String.format(Locale.ROOT, "https://%s.dfs.core.windows.net", accountName);
+
+        /*
+         * Create a DataLakeServiceClient object that wraps the service endpoint, credential and a request pipeline.
+         */
+        DataLakeServiceClient storageClient = new DataLakeServiceClientBuilder().endpoint(endpoint).credential(credential).buildClient();
+
+        /*
+         * This example shows several common operations just to get you started.
+         */
+
+        /*
+         * Create a client that references a to-be-created file system in your Azure Storage account. This returns a
+         * FileSystem object that wraps the file system's endpoint, credential and a request pipeline (inherited from storageClient).
+         * Note that file system names require lowercase.
+         */
+        FileSystemClient fileSystemClient = storageClient.getFileSystemClient("myjavafilesystembasic" + System.currentTimeMillis());
+
+        /*
+         * Create a file system in Storage datalake account.
+         */
+        fileSystemClient.create();
+
+        /*
+         * Create a client that references a to-be-created file in your Azure Storage account's file system.
+         * This returns a DataLakeFileClient object that wraps the file's endpoint, credential and a request pipeline
+         * (inherited from fileSystemClient). Note that file names can be mixed case.
+         */
+        DataLakeFileClient fileClient = fileSystemClient.getFileClient("HelloWorld.txt");
+
+        String data = "Hello world!";
+        InputStream dataStream = new ByteArrayInputStream(data.getBytes(StandardCharsets.UTF_8));
```

Comment:
Can you add in creating a directory client and putting a file under it? That way we show all the basic client types.

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/6106